### PR TITLE
Integrate `wc_analytics/customers` endpoint from fluxc to the app

### LINKED ISSUE #9211: Integrate `wc_analytics/customers` from fluxc to the app

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #9211

### Description

The PR:
* Replaces usage of the `/customers` endpoint to `wc_analytics/reports/customers`
* The new endpoint also returns unregistered customers. We use the data it gives for the unregistered customers and fetch all the data for registered
* It removes 3 chars thresholds to start the search
* It adds a few simple error indications as the original code doesn't have any
* Removes avatars from the UI as the new UI doesn't return them

Merge after https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2751 is merged

### Testing instructions

* Start order creation
* Click "add customer"
* Search for both registered and unregistered customers. Use their data

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/5a020c4c-dfac-4360-8f3e-ee92db04167f

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListViewModel.kt [hunk 1]
```diff
@@ -45,48 +45,67 @@ class CustomerListViewModel @Inject constructor(
...
-                    firstName = wcCustomer.billingFirstName,
-                    lastName = wcCustomer.billingLastName,
-                    country = wcCustomer.billingCountry,
-                    state = wcCustomer.billingState,
-                    postcode = wcCustomer.billingPostcode,
-                    phone = wcCustomer.billingPhone,
-                    email = wcCustomer.billingEmail
-                )
+            if (customerRemoteId != 0L) {
+                _viewState.value = _viewState.value?.copy(isSkeletonShown = true)
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListViewModel.kt:49]: NP: ❓ We could write tests to verify
1. Shimmer is shown when we are fetching the customer list
2. Shimmer is hidden when the customer fetch is a success
3. Shimmer is hidden when the customer fetch is a failure
4. Snackbar event is triggered when the customer fetch is a failure
5. Snackbar event is triggered when the customer fetch is successful but the model is empty
6. Snackbar event is triggered when the customer id is 0L and is a failure
7. Search is triggered even on a single character (as opposed to 3 characters previously)
8. CustomerSelected event is triggered when the customer fetch is successful.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListRepository.kt [hunk 2]
```diff
@@ -42,12 +43,18 @@ class CustomerListRepository @Inject constructor(
...
+            page = 1,
         ).takeUnless { it.isError }?.model
 
-    fun getCustomerByRemoteId(remoteId: Long): WCCustomerModel? =
-        customerStore.getCustomerByRemoteId(selectedSite.get(), remoteId)
+    suspend fun fetchCustomerByRemoteId(remoteId: Long): WooResult<WCCustomerModel> {
+        return customerStore.fetchSingleCustomer(selectedSite.get(), remoteId)
+    }
+
+    fun getCustomerByRemoteIdFromLocalStorage(remoteId: Long): WCCustomerModel? {
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListRepository.kt]: ❓ **No need to change anything:** I think we follow a convention that a method name prefixed with `fetch` indicates that the data is retrieved from the server and for any method with the prefix `get`, the data would be from the local storage. 

I have no problem with the method name `getCustomerByRemoteIdFromLocalStorage`. I just wanted to bring this up. 

In my mind, as soon as I read `fetchCustomerByRemoteId` I assume the data is "fetched" from the server, and for `getCustomerByRemoteId`, the data is retrieved from the local storage.

But better be explicit than any assumption. 👍

**Review by AnirudhBhat — APPROVED** [review]
Thank you for working on this @kidinov. The feature works as expected!

I've left couple of minor comments that are non-blocking for the merge.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListViewModel.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListViewModel.kt:49]: I added some testes d923b07eda807494fb98252e4aba8847f1009aad

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListRepository.kt (hunk 2 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/creation/customerlist/CustomerListRepository.kt]: I think consistency is better so renamed that to `getCustomerByRemoteId`. Thanks!

**Pull request merged**